rustls/tests/common/mod.rs

```diff
@@ -28,10 +29,14 @@ use rustls::{
     DigitallySignedStruct, DistinguishedName, Error, InconsistentKeys, NamedGroup, ProtocolVersion,
     RootCertStore, ServerConfig, ServerConnection, SideData, SignatureScheme, SupportedCipherSuite,
 };
+
 use webpki::anchor_from_trusted_cert;
 
 use super::provider;
 
+// Import `Arc` here for tests - can be overwritten to test with another Arc  such as `portable_atomic_util::Arc`
+pub use std::sync::Arc;
```

Comment:
It feels awkward that the concrete `Arc` type is selected separately for the library and the tests. It doesn't feel like there's a case where you want to change out one and not the other.

I imagine potential solutions to that have their own downsides. WDYT?

Reply:
Hmm ... my thought is that tests would have to follow the library API ... fork changes the library API then may need to update the tests.

I recall earlier discussion of adding `internal::alias` to export an undocumented alias & help alleviate this issue with the tests. This was my initial approach, changed as discussed earlier here: https://github.com/rustls/rustls/pull/2285#discussion_r1891378527